cms/search/tests/test_publishers.py

```diff
@@ -134,6 +134,22 @@ def test_publish_deleted(self, mock_producer_class):
 
         mock_future.get.assert_called_once_with(timeout=10)
 
+    def test_token_provider_cache_key(self):
+        token_provider = IAMKafkaTokenProvider()
+        self.assertEqual(
+            token_provider.token.get_cache_key(token_provider), "cms.search.publishers.IAMKafkaTokenProvider"
+        )
+
+    @patch("cms.search.publishers.MSKAuthTokenProvider.generate_auth_token")
+    def test_token_provider(self, mock_generate_auth_token):
+        mock_generate_auth_token.return_value = ("msk-token", None)
+        token_provider = IAMKafkaTokenProvider()
+
+        self.assertEqual(token_provider.token(), "msk-token")
+        self.assertEqual(token_provider.token(), "msk-token")
+
+        mock_generate_auth_token.assert_called_once()
```

Comment:
Nit: Could we do this just for greater test clarity?
```suggestion
        mock_generate_auth_token.assert_not_called()

        # Retrieve the token twice to ensure caching works
        self.assertEqual(token_provider.token(), "msk-token")
        self.assertEqual(token_provider.token(), "msk-token")

        mock_generate_auth_token.assert_called_once()
```